search/engine.go

```diff
@@ -360,10 +437,10 @@ func compareHits(h1 *elastic.SearchHit, h2 *elastic.SearchHit, sortBy string) (b
 
 func joinResponses(r1 *elastic.SearchResult, r2 *elastic.SearchResult, sortBy string, from int, size int) (*elastic.SearchResult, error) {
 	log.Infof("%+v %+v", r1, r2)
-	if r1.Hits.TotalHits == 0 {
+	if r1 == nil || r1.Hits == nil || r1.Hits.TotalHits == 0 {
```

Comment:
Create function similar to haveHits that is called noHits and use it here.

Reply:
using !haveHits